projects/components/src/select/select.component.ts

```diff
@@ -204,6 +210,27 @@ export class SelectComponent<V> implements AfterContentInit, OnChanges {
     }
   }
 
+  public getPrefixIcon(
+    selectedOption: SelectOptionComponent<V> | undefined,
+    triggerIcon: string | undefined
+  ): string | undefined {
+    if (selectedOption !== undefined && selectedOption?.icon !== undefined) {
```

Comment:
So we're favoring the selections icon in the trigger over the trigger's icon? that doesn't seem right to me, but I guess I'd also find it strange to use a trigger icon and individual icons together

Reply:
(also, unnecessary first condition)
```suggestion
    if (selectedOption?.icon !== undefined) {
```

Reply:
if there's a selection, we should show the selection's icon or nothing.
If there's no selection, show the trigger icon. That's the intent.

It doesn't make sense to show the regular trigger icon along with a selection. that would be confusing.

we already show selected label in the trigger the same way. This is extending it to the icon as well. 

Fixed the logic.

Reply:
I think my brain wasn't functioning last night when I wrote the logic for this. 🤔

Reply:
Using the selected icon if defined makes sense, but not using the component wide icon if the selections aren't individually... iconized breaks the original intent. I think it's just two different styles:
![image](https://user-images.githubusercontent.com/45047841/119974354-63b1cc80-bf69-11eb-9d13-adf4b97a8e50.png)

in this one, that icon just is meant to apply to every option, but not appear as a duplicate in the option list.

Reply:
updated